<launch>

  <arg name="custom_config" default="" />

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(env UAV_NAME)" />
  <arg name="RUN_TYPE" default="$(env RUN_TYPE)" />
  <arg name="KML_FILE_PATH" default="" />
  <arg name="SIMULATION_START_LAT" default="49.541502"/>
  <arg name="SIMULATION_START_LONG" default="15.348656"/>
  <arg name="ALTITUDE" default="6.0"/>
  <arg name="ENERGY_MODEL_CONFIG_FILE" default = "$(find thesis_path_generator)/custom_configs/energy_model_config.yaml"/>

  <!-- set to false if the nodelet should run under an external nodelet manager -->
  <arg name="standalone" default="true" />

  <!-- allows disabling the gui from command line -->
  <arg name="gui" default="true" />

  <!-- will it run using GNU debugger? -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="manager" default="$(arg UAV_NAME)_thesis_path_generator_manager" />
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>

  <!-- Namespace - all topics and parameters will be remapped using this namespace as a prefix (eg.: "/vision_example/image_in" to "/uav1/vision_example/image_in") -->
  <group ns="$(arg UAV_NAME)">

    <!-- TrajectoryGenerator nodelet -->
    <node pkg="nodelet" type="nodelet" name="path_generation" args="$(arg nodelet) path_generation/PathGenerator $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <!-- Parameters loaded from launch file -->
      <param name="gui" type="bool" value="$(arg gui)"/>

      <param name="UAV_NAME" type="string" value="$(arg UAV_NAME)"/>
      <param name="KML_FILE_PATH" type="string" value="$(arg KML_FILE_PATH)"/>
      <param name="SIMULATION" type="bool" value="true"/>
      <param name="SIMULATION_START_LAT" type="double" value="$(arg SIMULATION_START_LAT)"/>
      <param name="SIMULATION_START_LONG" type="double" value="$(arg SIMULATION_START_LONG)"/>
      <param name="ALTITUDE" type="double" value="$(arg ALTITUDE)"/>


      <!-- ROS parameters config file -->
      <rosparam file="$(find thesis_path_generator)/config/$(arg RUN_TYPE).yaml" />
      <rosparam if="$(eval not arg('custom_config') == '')" file="$(arg custom_config)" />
      <rosparam file="$(arg ENERGY_MODEL_CONFIG_FILE)" />

      <!-- Subscribers (note that the topic names will also be remapped using the namespace as a prefix) -->

      <!-- Publishers (note that the topic names will also be remapped using the namespace as a prefix) -->

    </node>

  </group>

</launch>
